src/VSCodeExtension/src/formatter/rules/indent.ts

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { FormatRule } from "../formatter";
+
+/**
+ * Removes whitespace from left of `namespace`.
+ *
+ * @param code input string
+ *
+ * @example `   namespace   Foo   {}` ->
+ * `namespace Foo {}`
+ */
+export const namespaceRule: FormatRule = (code: string) => {
+    const namespaceMatcher: RegExp = /^\s*namespace\s*(\w+)\s*(\S|\S.*\S)\s*$/g;

```

Comment:
Does `\w+` catch dotted names like `Microsoft.Quantum.Arrays`?

Reply:
Good catch. What do you think of this:

```javascript
/^\s*namespace\s+(\w+(?:\.\w+)*)\s*({|{.*})\s*/$
```

Reply:
Please make a test for every edge case you encounter